presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/thrift/ThriftHiveMetastore.java

```diff
@@ -259,10 +265,11 @@ public ThriftMetastoreStats getStats()
         }
     }
 
-    private Table getTableFromMetastore(String databaseName, String tableName)
+    private Table getTableFromMetastore(ClientSupplier supplier, String databaseName, String tableName)
```

Comment:
Pass `HiveIdentity` here. The other call where you use different supplier is unnecessary.